### PR TITLE
Documentation for chashmap library

### PR DESCRIPTION
**Pull request by Maksasj**

### CODE REVIEW

File: include/chashmap/chashmap.h [hunk 1]
```diff
@@ -6,70 +36,83 @@
...
+ * @param   map         Actual pointer to hashmap
+*/
 void mapInit(Hashmap* map);
 
+/**
+ * Inserts new value to the hashmap
+ * 
+ * @param   map         Pointer to hashmap
+ * @param   key         Key where to insert the value
+ * @param   value       Attributes of new children node
```
Comment (AlexShukel) [inline include/chashmap/chashmap.h]: bad description

File: include/chashmap/chashmap.h [hunk 2]
```diff
@@ -6,70 +36,83 @@
...
 HashmapEntry* mapInsert(Hashmap* map, char* key, char* value);
 
+/**
+ * Returns pointer to value of entry located at this key.
+ * 
+ * @param   map         Pointer to hashmap
+ * @param   key         Key where expected entry is located
+ * @return  Pointer to value or NULL if entry with provided key failed to find
+*/
 void* mapGetValue(Hashmap* map, char* key);
```
Comment (AlexShukel) [inline include/chashmap/chashmap.h]: This must return char *, as you declared in HashmapEntry struct

File: include/chashmap/chashmap.h [hunk 3]
```diff
@@ -6,70 +36,83 @@
 #include <stdlib.h>
 #include <string.h>
 
+/* Structre what contains */
```
Comment (AlexShukel) [inline include/chashmap/chashmap.h]: Unnecessary comment

File: include/chashmap/chashmap.h [hunk 4]
```diff
@@ -1,3 +1,33 @@
...
+        mapInsert(&hm, "key2", "value2");
+
+        char* helloWorld = mapGetValue(&hm, "key1");
+
+        if(mapKeyExist(&hm, "key2"))
+            printf("Key 'key2' exist \n");
+
+        if(!mapKeyExist(&hm, "randomkey"))
+            printf("Key randomkey does not exist \n");
+
```
Comment (AlexShukel) [inline include/chashmap/chashmap.h:25]: Add mapFree in example

File: include/chashmap/chashmap.h [hunk 5]
```diff
@@ -6,70 +36,83 @@
...
+ * Returns next entry of the hashmap,
+ * Was conceived in order to iterate over all entries of the hashmap
+ * 
+ * @param   node    Pointer to hashmap
+ * @return  Returns pointer to next entry of the hashmap
+*/
 HashmapEntry* getNextEntry(Hashmap* map);
 
+// Helper functions
+unsigned int chashmapHash(const char* key);
```
Comment (AlexShukel) [inline include/chashmap/chashmap.h]: Move these functions to private API (create separate header for them in src directory), as they are not supposed to be used by user.

**Review by AlexShukel — COMMENTED** [review]
Some notes

File: include/chashmap/chashmap.h (hunk 1 above)
Comment (Maksasj) [reply include/chashmap/chashmap.h]: u clown

**Review by AlexShukel — APPROVED** [review]
Ок

**Pull request merged**